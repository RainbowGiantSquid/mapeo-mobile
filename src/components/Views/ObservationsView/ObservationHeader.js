// @flow
import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';
import { LIGHT_GREY } from '../../../lib/styles';

type Props = {
  title: string
};

const styles = StyleSheet.create({
  container: {
    height: 50,
    flexDirection: 'row',
    backgroundColor: 'white',
    justifyContent: 'flex-start',
    alignItems: 'center',
    borderBottomColor: LIGHT_GREY,
    borderBottomWidth: 1,
    width: Dimensions.get('window').width - 30,
    paddingHorizontal: 30
  },
  title: {
    fontSize: 18,
    fontWeight: '700',
    lineHeight: 23,
    color: 'black'
  }
});

const ObservationHeader = (props: Props) => (
  <View style={styles.container}>
    <Text style={styles.title}>{props.title}</Text>
  </View>
);

export default ObservationHeader;
