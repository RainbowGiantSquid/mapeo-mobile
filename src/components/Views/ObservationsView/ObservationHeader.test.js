// @flow
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ObservationHeader from './ObservationHeader';

describe('ObservationHeader tests', () => {
  const closeRightDrawer = jest.fn();

  beforeEach(() => {
    closeRightDrawer.mockReset();
  });

  test('snapshots', () => {
    let tree;
    tree = renderer.create(
      <ObservationHeader closeRightDrawer={closeRightDrawer} showSyncTip />
    );
    expect(tree).toMatchSnapshot();
  });
});
