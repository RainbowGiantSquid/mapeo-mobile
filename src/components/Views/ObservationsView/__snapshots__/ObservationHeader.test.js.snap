// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationHeader tests snapshot with showSyncTip as false 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d6d2cf",
            "borderRadius": 50,
            "height": 35,
            "justifyContent": "center",
            "opacity": 1,
            "width": 35,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#EAEAEA",
              "borderRadius": 50,
              "height": 30,
              "justifyContent": "center",
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "15deg",
                    },
                  ],
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "borderBottomWidth": 1,
        "borderColor": "#EAEAEA",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      All Observations
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      View by Most Recent
    </Text>
  </View>
</View>
`;

exports[`ObservationHeader tests snapshot with showSyncTip as true 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 65,
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d6d2cf",
            "borderRadius": 50,
            "height": 35,
            "justifyContent": "center",
            "opacity": 1,
            "width": 35,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#EAEAEA",
              "borderRadius": 50,
              "height": 30,
              "justifyContent": "center",
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                },
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "15deg",
                    },
                  ],
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#a5a5a4",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "borderBottomWidth": 1,
        "borderColor": "#EAEAEA",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      All Observations
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      View by Most Recent
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "elevation": 3,
        "justifyContent": "center",
        "position": "absolute",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 100,
          "width": 0,
        },
        "shadowOpacity": 6,
        "shadowRadius": 100,
        "top": 50,
        "zIndex": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#3366ff",
          "borderRadius": 8,
          "flexDirection": "row",
          "height": 35,
          "marginTop": 7,
          "paddingHorizontal": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "marginHorizontal": 20,
          }
        }
      >
        Connect to sync
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomColor": "#3366ff",
          "borderBottomWidth": 7,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 7,
          "borderRightColor": "transparent",
          "borderRightWidth": 7,
          "borderStyle": "solid",
          "height": 0,
          "position": "absolute",
          "top": 0,
          "width": 0,
        }
      }
    />
  </View>
</View>
`;
